README.md

```diff
@@ -1 +1,569 @@
+	"rsc.io/goversion/version"
+)
+```
+
+Divide imports into four groups sorted from internal to external for readability:
+* Standard library
+* Project internal packages
+* Company internal packages
+* External packages
+
+
+## In-Band Errors
+
+In C and similar languages, it's common for functions to return values like -1 or null to signal errors or missing results:
+
+```Go:
+// Lookup returns the value for key or "" if there is no mapping for key.
+func Lookup(key string) string
+
+// Failing to check a for an in-band error value can lead to bugs:
+Parse(Lookup(key))  // returns "parse failure for value" instead of "no value for key"
+```
+
+Go's support for multiple return values provides a better solution. Instead of requiring clients to check for an in-band error value, a function should return an additional value to indicate whether its other return values are valid. This return value may be an error, or a boolean when no explanation is needed. It should be the final return value.
+
+```Go:
+// Lookup returns the value for key or ok=false if there is no mapping for key.
+func Lookup(key string) (value string, ok bool)
+```
+
+This prevents the caller from using the result incorrectly:
+
+```Go:
+Parse(Lookup(key))  // compile-time error
+```
+
+And encourages more robust and readable code:
+
+```Go:
+value, ok := Lookup(key)
+if !ok  {
+    return fmt.Errorf("no value for %q", key)
+}
+return Parse(value)
+```
+
+This rule applies to exported functions but is also useful for unexported functions.
+
+Return values like nil, "", 0, and -1 are fine when they are valid results for a function, that is, when the caller need not handle them differently from other values.
+
+Some standard library functions, like those in package "strings", return in-band error values. This greatly simplifies string-manipulation code at the cost of requiring more diligence from the programmer. In general, Go code should return additional values for errors.
+
+## Initialisms
+
+Words in names that are initialisms or acronyms (e.g. "URL" or "NATO") have a consistent case. For example, "URL" should appear as "URL" or "url" (as in "urlPony", or "URLPony"), never as "Url". As an example: ServeHTTP not ServeHttp. For identifiers with multiple initialized "words", use for example "xmlHTTPRequest" or "XMLHTTPRequest".
+
+This rule also applies to "ID" when it is short for "identifier" (which is pretty much all cases when it's not the "id" as in "ego", "superego"), so write "appID" instead of "appId".
+
+Code generated by the protocol buffer compiler is exempt from this rule. Human-written code is held to a higher standard than machine-written code.
+
+## Interfaces
+
+Go interfaces generally belong in the package that uses values of the interface type, not the package that implements those values. The implementing package should return concrete (usually pointer or struct) types: that way, new methods can be added to implementations without requiring extensive refactoring.
```

Comment:
Counterfeiter disagrees with this sentence. We should amend it or figure out how to work around counterfeiter restrictions.